include/imgproc/mask.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+// Third-party includes
+#include "third_party/path.h"
+
+// OpenCV
+#include <opencv2/opencv.hpp>
+
+// Standard C includes
+#include <cstddef>
+
+namespace BoBRobotics
+{
+namespace ImgProc
+{
+//! A class providing extra checks and type safety for image masks
+class Mask
+{
+public:
+    //! Create an empty mask (i.e. treat all pixels as unmasked)
+    Mask() = default;
+
+    //! Construct a mask based on a cv::Mat where zero pixels are treated as masked
+    explicit Mask(cv::Mat mask, const cv::Size &resizeTo = {});
```

Comment:
I *think* this is the right way to pass objects into setters/constructors (there is actually a clang-tidy option to check for this which I didn't enable because it seems a bit overzealous).

The idea is that if you then move the object into the member variable, you can avoid a copy in the case that ``mask`` is passed as an rvalue.